File: api/functional/flow-creator/src/main/java/com/emanuelvictor/api/functional/flowcreator/FlowCreatorApplication.java
package com.emanuelvictor.api.functional.flowcreator;

import com.emanuelvictor.api.functional.flowcreator.infrastructure.helprs.PopulateHelper;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.builder.SpringApplicationBuilder;
import org.springframework.boot.context.event.ApplicationReadyEvent;
import org.springframework.boot.web.servlet.support.SpringBootServletInitializer;
import org.springframework.context.ApplicationListener;
import org.springframework.context.annotation.Bean;
import org.springframework.scheduling.annotation.EnableAsync;

/**
 *
 */
@SpringBootApplication
public class FlowCreatorApplication extends SpringBootServletInitializer {

    /**
     *
     */
    @Autowired
    PopulateHelper populateHelper;

    /**
     * @param args String[]
     */
    public static void main(final String[] args) {
        SpringApplication.run(FlowCreatorApplication.class, args);
    }

    /**
     * @param application SpringApplicationBuilder
     * @return SpringApplicationBuilder
     */
    @Override
    protected SpringApplicationBuilder configure(final SpringApplicationBuilder application) {
        return application.sources(FlowCreatorApplication.class);
    }

    /**
     * @return ApplicationListener<ApplicationReadyEvent>
     */
    @Bean
    public ApplicationListener<ApplicationReadyEvent> getApplicationReadyEvent() {
        return applicationReadyEvent -> {

            populateHelper.eraseData();
            populateHelper.populateData();

            populateHelper.startProgram();
        };
    }

}

File: api/functional/flow-creator/src/main/java/com/emanuelvictor/api/functional/flowcreator/FlowCreatorApplication.kt
package com.emanuelvictor.api.functional.flowcreator

import com.emanuelvictor.api.functional.flowcreator.domain.services.AlternativeService
import com.emanuelvictor.api.functional.flowcreator.domain.services.ChoiceService
import com.emanuelvictor.api.functional.flowcreator.infrastructure.adapters.AlternativeRepositoryImpl
import com.emanuelvictor.api.functional.flowcreator.infrastructure.adapters.ChoiceRepositoryImpl
import com.emanuelvictor.api.functional.flowcreator.infrastructure.adapters.OptionRepositoryImpl
import com.emanuelvictor.api.functional.flowcreator.infrastructure.helprs.PopulateHelper

/**
 *
 */
object FlowCreatorApplication {

    /**
     * @param args String[]
     */
    @JvmStatic
    fun main(args: Array<String>) {

        val optionRepository = OptionRepositoryImpl();
        val choiceRepository = ChoiceRepositoryImpl();
        val choiceService = ChoiceService(choiceRepository)
        val alternativeRepository = AlternativeRepositoryImpl()
        val alternativeService = AlternativeService(alternativeRepository);
        val populateHelper = PopulateHelper(choiceService, optionRepository, choiceRepository, alternativeService)

        populateHelper.populateData();
        populateHelper.startProgram();
    }
}